compiler/noirc_frontend/src/hir/type_check/stmt.rs

```diff
@@ -265,8 +266,17 @@ impl<'interner> TypeChecker<'interner> {
                     expr_span,
                 }
             });
-            if annotated_type.is_unsigned() {
-                self.lint_overflowing_uint(&rhs_expr, &annotated_type);
+            if let Some(bit_size) = annotated_type.bit_size() {
+                let max_integer_bit_size = FieldElement::max_num_bits() / 2;
+                if bit_size > max_integer_bit_size {
+                    self.errors.push(TypeCheckError::UnsupportedIntegerSize {
+                        num_bits: bit_size,
+                        max_num_bits: max_integer_bit_size,
+                        span: expr_span,
+                    });
```

Comment:
Why do we only check this on a variable declaration? And why do we allow u128 to get all the way to typechecking at all if it is unsupported? Can we just error in the parser or even lexer instead?

Reply:
Good call. Turns out we have a max bits check in the lexer already but we just check against the FieldElement max bits rather than `FieldElement::max_num_bits / 2`. I ended up just dividing the max bits by 2.